src/ziggurat/kafka_consumer/consumer_driver.clj

```diff
@@ -0,0 +1,53 @@
+(ns ziggurat.kafka-consumer.consumer-driver
+  (:require [mount.core :refer [defstate]]
+            [ziggurat.kafka-consumer.consumer :as ct]
+            [ziggurat.kafka-consumer.consumer-handler :as ch]
+            [ziggurat.config :refer [ziggurat-config]]
+            [clojure.tools.logging :as log]
+            [ziggurat.kafka-consumer.executor-thread-pool :refer :all]
+            [mount.core :as mount]
+            [ziggurat.metrics :as metrics])
+  (:import (java.util.concurrent ExecutorService RejectedExecutionException)
+           (org.apache.kafka.clients.consumer Consumer)))
+
+(defn- start-polling-with-consumer
+  [consumer init-arg topic-entity consumer-config]
+  (let [message-poller (cast Runnable #(ch/poll-for-messages consumer (:handler-fn init-arg) topic-entity consumer-config))]
+    (when message-poller
+      (try
+        (.submit ^ExecutorService executor-thread-pool ^Runnable message-poller)
+        (catch RejectedExecutionException e
+          (metrics/increment-count ["ziggurat.batch.consumption"] "thread-pool.task.rejected" 1 {:topic-entity topic-entity})
+          (log/error "message polling task was rejected by the threadpool" e))))))
+
+(defn- create-consumers-per-group
```

Comment:
I think this function is doing more than the name suggests. It is creating and starting polling both.